### PR TITLE
Bump safe-eth-py and refactor for Web3 v6

### PR DESCRIPTION
**Pull request by Uxio0**
- Bump safe-eth-py and modify the required tests for Web3 v6
- Take a look at https://web3py.readthedocs.io/en/stable/v6_migration.html#migrating-v5-to-v6

### CODE REVIEW

File: safe_transaction_service/history/models.py [hunk 1]
```diff
@@ -310,6 +310,7 @@ def create_from_tx_dict(
         tx_receipt: Optional[Dict[str, Any]] = None,
         ethereum_block: Optional[EthereumBlock] = None,
     ) -> "EthereumTx":
+        print(tx, tx_receipt)
```
Comment (moisses89) [inline safe_transaction_service/history/models.py:313]: Ups :)

File: safe_transaction_service/history/models.py [hunk 2]
```diff
@@ -338,7 +339,7 @@ def create_from_tx_dict(
             nonce=tx["nonce"],
             to=tx.get("to"),
             value=tx["value"],
-            type=int(tx.get("type", "0x0"), 0),
+            type=tx.get("type", 0),
```
Comment (moisses89) [inline safe_transaction_service/history/models.py:342]: Type is int in the model of `ethereumTx`, tx["type"] is always int?

File: safe_transaction_service/history/tests/test_models.py [hunk 3]
```diff
@@ -289,7 +289,7 @@ def test_create_from_tx_dict(self):
             with self.subTest(tx_mock=tx_mock):
                 tx_dict = tx_mock["tx"]
                 ethereum_tx = EthereumTx.objects.create_from_tx_dict(tx_dict)
-                self.assertEqual(ethereum_tx.type, int(tx_dict["type"], 0))
+                self.assertEqual(ethereum_tx.type, tx_dict["type"], 0)
```
Comment (moisses89) [inline safe_transaction_service/history/tests/test_models.py:292]: The same as above

File: safe_transaction_service/history/utils.py [hunk 4]
```diff
@@ -43,9 +43,10 @@ def clean_receipt_log(receipt_log: Dict[str, Any]) -> Optional[Dict[str, Any]]:
     :param receipt_log:
     :return:
     """
+
     parsed_log = {
         "address": receipt_log["address"],
-        "data": receipt_log["data"],
+        "data": receipt_log["data"].hex(),
```
Comment (moisses89) [inline safe_transaction_service/history/utils.py:49]: data was before python bytes?

File: safe_transaction_service/history/models.py (hunk 2 above)
Comment (Uxio0) [reply safe_transaction_service/history/models.py:342]: Yes, now it's parsed by web3.py (before it was an `HexStr`)

File: safe_transaction_service/history/tests/test_models.py (hunk 3 above)
Comment (Uxio0) [reply safe_transaction_service/history/tests/test_models.py:292]: Same

File: safe_transaction_service/history/utils.py (hunk 4 above)
Comment (Uxio0) [reply safe_transaction_service/history/utils.py:49]: Data was an `HexStr` too, now it's parsed

**Review by moisses89 — APPROVED** [review]
Looks good to me!

**Pull request merged**